pkg/cloudprovider/aws/builder.go

```diff
@@ -40,3 +42,11 @@ func NewCloudProvider(logger logr.Logger) (cloudprovider.CloudProvider, error) {
 
 	return p, nil
 }
+
+// NewMockCloudProvider returns a new mock AWS cloud provider
+func NewMockCloudProvider(autoscalingiface autoscalingiface.AutoScalingAPI, ec2iface ec2iface.EC2API) cloudprovider.CloudProvider {
```

Comment:
Is this really returning a new mock AWS cloud provider or just a new provider that could either be the real thing or mocked depending on what is passed in?

Reply:
Good point, it's not necessarily mocking.